File: go/sudoku-solver/sudoku-solver.go
func solveSudoku(board [][]byte)  {
    solve(board)
}

func solve(board [][]byte) bool {
    for i := 0; i < len(board); i ++ {
        for j := 0; j < len(board[i]); j++ {
            if board[i][j] == byte('.') {
                for c := byte('1'); c <= byte('9'); c++ {
                    if isValid(board, i, j, c) {
                        board[i][j] = c
                        
                        if solved := solve(board); solved {
                            return true
                        } else {
                            board[i][j] = byte('.')
                        }
                    }
                }
                return false
            }
        }
    }
    return true
}

func isValid(board [][]byte, row int, col int, c byte) bool {
    for i := 0; i < 9; i++ {
        if board[i][col] == c ||board[row][i] == c || board[3 * (row/3) + i / 3][3 * (col / 3) + i % 3] == c{
            return false
        }
    }
    return true
}

File: js/sudoku-solver/sudoku-solver.js
var solveSudoku = function(board) {
    solve(board);
};

function isValid(board, row, col, num) {
    for (let i = 0; i < 9; i++) {
        if (board[row][i] === num || board[i][col] === num ||
            board[3 * Math.floor(row / 3) + Math.floor(i / 3)][3 * Math.floor(col / 3) + i % 3] === num) {
            return false;
        }
    }
    return true;
}

function solve(board) {
    for (let row = 0; row < 9; row++) {
        for (let col = 0; col < 9; col++) {
            if (board[row][col] === ".") {
                for (let num = 1; num <= 9; num++) {
                    if (isValid(board, row, col, num.toString())) {
                        board[row][col] = num.toString();
                        if (solve(board)) {
                            return true;
                        }
                        board[row][col] = ".";
                    }
                }
                return false;
            }
        }
    }
    return true;
}
